cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/connector/resource/AzureDatabaseResourceService.java

```diff
@@ -150,24 +155,34 @@
                 .build(), ResourceStatus.DELETED));
     }
 
-    private List<CloudResourceStatus> deleteDatabaseServer(List<CloudResource> resources, CloudContext cloudContext, boolean force,
+    private List<CloudResourceStatus> deleteResources(List<CloudResource> resources, CloudContext cloudContext, boolean force,
             AzureClient client, PersistenceNotifier persistenceNotifier) {
-        LOGGER.debug("Deleting database server");
-        Optional<CloudResource> dbServerResourceOptional = resources.stream().filter(r -> AZURE_DATABASE.equals(r.getType())).findFirst();
-        if (dbServerResourceOptional.isEmpty()) {
-            LOGGER.warn("Azure database id not found in database, deleting nothing");
-            return List.of();
-        }
-        String databaseServerId = dbServerResourceOptional.get().getReference();
-        Optional<String> errorMessage = azureUtils.deleteDatabaseServer(client, databaseServerId, force);
-        if (errorMessage.isEmpty()) {
-            deleteResources(resources, cloudContext, persistenceNotifier);
-        }
 
-        return Lists.newArrayList(new CloudResourceStatus(CloudResource.builder()
+        // TODO simplify after final form of template is reached
+
+        List<CloudResource> azureGenericResources = findResources(resources, List.of(AZURE_PRIVATE_ENDPOINT));
+        LOGGER.debug("Deleting dns zone groups and azure private endpoints {}", azureGenericResources);
+        azureUtils.deleteGenericResources(client, azureGenericResources.stream().map(CloudResource::getReference).collect(Collectors.toList()));
+        azureGenericResources.forEach(cr -> persistenceNotifier.notifyDeletion(cr, cloudContext));
+
+        return findResources(resources, List.of(AZURE_DATABASE)).stream()
+                .map(r -> deleteDatabaseServerAndNotify(r, cloudContext, client, persistenceNotifier, force))
+                .collect(Collectors.toList());
+    }
+
+    private CloudResourceStatus deleteDatabaseServerAndNotify(
+            CloudResource r, CloudContext cloudContext, AzureClient client, PersistenceNotifier persistenceNotifier, boolean force) {
```

Comment:
can you rename r to cloudResource? Not a big thing but helps reading it.

Reply:
done